### PR TITLE
Feature award data archive

### PR DESCRIPTION
**Pull request by ebdabbs**
- [x] [Backend](https://github.com/fedspendingtransparency/usaspending-api/pull/816) merged
- [x] Code review
- [x] Design review

- Implements the Award Data Archive page at `/bulk_download/award_data_archive`
- Adds an info box to the Awards section of the Bulk Download Page *hidden until script is run to populate the monthly files

https://federal-spending-transparency.atlassian.net/browse/DS-2025

### CODE REVIEW

File: src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            filters: {
+                agency: {
+                    id: 'all',
+                    name: 'All'
+                },
+                type: {
+                    name: 'contracts',
+                    display: 'Contracts'
+                },
+                fy: '2018'
```
Comment (kevinli-work) [inline src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx]: Can we utilize our FY helper to either generate a current fiscal year by default or put a hardcoded value in the fiscalYearHelper with the other hardcoded years?

File: src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        }
+        else {
+            this.resultsRequest = BulkDownloadHelper.requestArchiveFiles({
+                agency: parseInt(this.state.filters.agency.id, 10),
+                fiscal_year: parseInt(this.state.filters.fy, 10),
+                type: this.state.filters.type.name
+            });
+        }
+
+        this.agencyListRequest.promise
```
Comment (kevinli-work) [inline src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx]: It looks like you're attaching to the wrong request

File: src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+        data.forEach((item) => {
+            // Format Agency String
+            let formattedAgency = item.agency_name;
+            if (item.agency_acronym) {
+                formattedAgency = `${item.agency_name} (${item.agency_acronym})`;
+            }
+
+            // Format Updated Date
+            const date = moment(item.updated_date);
```
Comment (kevinli-work) [inline src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx]: Can you specify the parse input format? This helps avoid ambiguities (ie, what is the month in 2017-04-03) https://momentjs.com/docs/#/parsing/string-format/

File: src/js/components/bulkDownload/BulkDownloadPage.jsx [hunk 4]
```diff
@@ -67,6 +69,7 @@ export default class BulkDownloadPage extends React.Component {
 
     changeDataType(dataType) {
         this.props.setDataType(dataType);
+        Router.history.replace('/bulk_download');
```
Comment (kevinli-work) [inline src/js/components/bulkDownload/BulkDownloadPage.jsx]: See later comment about avoiding spreading routing logic around.

File: src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx (hunk 3 above)
Comment (ebdabbs) [reply src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx]: Confirmed the API is returning `YYYY-MM-DD`

File: src/js/components/bulkDownload/BulkDownloadPage.jsx (hunk 4 above)
Comment (ebdabbs) [reply src/js/components/bulkDownload/BulkDownloadPage.jsx]: This removes `/award_data_archive` from the url when a user clicks an item in the sidebar

**Pull request merged**